File: src/cli/apt.go
package cli

import (
	"fmt"
	"strings"
)

// Apt wraps APT/APT-GET commands with fluent API
type Apt struct {
	quiet              bool
	useAptGet          bool
	noInstallRecommends bool
	options            map[string]string
}

// NewApt creates a new Apt command builder
func NewApt() *Apt {
	return &Apt{}
}

// Quiet sets quiet mode
func (a *Apt) Quiet() *Apt {
	a.quiet = true
	return a
}

// UseAptGet uses apt-get instead of apt
func (a *Apt) UseAptGet() *Apt {
	a.useAptGet = true
	return a
}

// NoInstallRecommends sets --no-install-recommends flag
func (a *Apt) NoInstallRecommends() *Apt {
	a.noInstallRecommends = true
	return a
}

// Options sets apt options (e.g., Dpkg::Options::)
func (a *Apt) Options(opts map[string]string) *Apt {
	if a.options == nil {
		a.options = make(map[string]string)
	}
	for k, v := range opts {
		a.options[k] = v
	}
	return a
}

// Update generates apt update command
func (a *Apt) Update() string {
	cmd := "apt"
	if a.useAptGet {
		cmd = "apt-get"
	}
	flags := []string{}
	if a.quiet {
		flags = append(flags, "-qq")
	}
	flagStr := ""
	if len(flags) > 0 {
		flagStr = " " + strings.Join(flags, " ")
	}
	return fmt.Sprintf("%s%s update", cmd, flagStr)
}

// Install generates apt install command
func (a *Apt) Install(packages []string) string {
	cmd := "apt"
	if a.useAptGet {
		cmd = "apt-get"
	}
	flags := []string{"-y"}
	if a.quiet {
		flags = append(flags, "-qq")
	}
	if a.noInstallRecommends {
		flags = append(flags, "--no-install-recommends")
	}
	
	// Build options (-o key=value)
	optParts := []string{}
	if a.options != nil {
		for key, value := range a.options {
			// For Dpkg::Options::, split multiple options and add each separately
			if key == "Dpkg::Options::" && strings.Contains(value, " ") {
				optionsList := strings.Fields(value)
				for _, opt := range optionsList {
					optParts = append(optParts, fmt.Sprintf("-o %s=%s", key, opt))
				}
			} else {
				optParts = append(optParts, fmt.Sprintf("-o %s=%s", key, value))
			}
		}
	}
	
	parts := []string{cmd}
	parts = append(parts, strings.Join(flags, " "))
	if len(optParts) > 0 {
		parts = append(parts, strings.Join(optParts, " "))
	}
	parts = append(parts, "install")
	parts = append(parts, strings.Join(packages, " "))
	return strings.Join(parts, " ")
}

// Upgrade generates apt upgrade command
func (a *Apt) Upgrade() string {
	cmd := "apt"
	if a.useAptGet {
		cmd = "apt-get"
	}
	flags := []string{"-y"}
	if a.quiet {
		flags = append(flags, "-qq")
	}
	return fmt.Sprintf("%s %s upgrade", cmd, strings.Join(flags, " "))
}

// DistUpgrade generates apt dist-upgrade command
func (a *Apt) DistUpgrade() string {
	cmd := "apt-get"
	flags := []string{"-y"}
	if a.quiet {
		flags = append(flags, "-qq")
	}
	return fmt.Sprintf("%s %s dist-upgrade", cmd, strings.Join(flags, " "))
}

// IsInstalledCheckCmd generates command to check if package is installed
func IsInstalledCheckCmd(packageName string) string {
	return fmt.Sprintf("dpkg -l | grep -q '^ii.*%s' && echo installed || echo not_installed", packageName)
}

// ParseIsInstalled parses output to check if package is installed
func ParseIsInstalled(output string) bool {
	return strings.Contains(output, "installed")
}

File: src/CLI/Apt.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Enva.CLI;

public class Apt
{
    private bool quiet;
    private bool useAptGet;
    private bool noInstallRecommends;
    private Dictionary<string, string> options = new Dictionary<string, string>();

    public static Apt NewApt()
    {
        return new Apt();
    }

    public Apt Quiet()
    {
        quiet = true;
        return this;
    }

    public Apt UseAptGet()
    {
        useAptGet = true;
        return this;
    }

    public Apt NoInstallRecommends()
    {
        noInstallRecommends = true;
        return this;
    }

    public Apt Options(Dictionary<string, string> opts)
    {
        foreach (var kvp in opts)
        {
            options[kvp.Key] = kvp.Value;
        }
        return this;
    }

    public string Update()
    {
        string cmd = useAptGet ? "apt-get" : "apt";
        var flags = new List<string>();
        if (quiet)
        {
            flags.Add("-qq");
        }
        string flagStr = flags.Any() ? " " + string.Join(" ", flags) : "";
        return $"{cmd}{flagStr} update";
    }

    public string Install(IEnumerable<string> packages)
    {
        string cmd = useAptGet ? "apt-get" : "apt";
        var flags = new List<string> { "-y" };
        if (quiet)
        {
            flags.Add("-qq");
        }
        if (noInstallRecommends)
        {
            flags.Add("--no-install-recommends");
        }
        
        var optParts = new List<string>();
        if (options != null && options.Any())
        {
            foreach (var kvp in options)
            {
                if (kvp.Key == "Dpkg::Options::" && kvp.Value.Contains(" "))
                {
                    var optionsList = kvp.Value.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
                    foreach (var opt in optionsList)
                    {
                        optParts.Add($"-o {kvp.Key}={opt}");
                    }
                }
                else
                {
                    optParts.Add($"-o {kvp.Key}={kvp.Value}");
                }
            }
        }
        
        var parts = new List<string> { cmd };
        parts.Add(string.Join(" ", flags));
        if (optParts.Any())
        {
            parts.Add(string.Join(" ", optParts));
        }
        parts.Add("install");
        parts.Add(string.Join(" ", packages));
        return string.Join(" ", parts);
    }

    public string Upgrade()
    {
        string cmd = useAptGet ? "apt-get" : "apt";
        var flags = new List<string> { "-y" };
        if (quiet)
        {
            flags.Add("-qq");
        }
        return $"{cmd} {string.Join(" ", flags)} upgrade";
    }

    public string DistUpgrade()
    {
        string cmd = "apt-get";
        var flags = new List<string> { "-y" };
        if (quiet)
        {
            flags.Add("-qq");
        }
        return $"{cmd} {string.Join(" ", flags)} dist-upgrade";
    }

    public static string IsInstalledCheckCmd(string packageName)
    {
        return $"dpkg -l | grep -q '^ii.*{packageName}' && echo installed || echo not_installed";
    }

    public static bool ParseIsInstalled(string output)
    {
        return output.Contains("installed");
    }
}
